api/python/quilt3/packages.py

```diff
@@ -956,45 +974,57 @@ def push(self, name, dest=None, registry=None, message=None):
         validate_package_name(name)
 
         if registry is None:
-            if dest is None:
-                # Only a package name is set, so set registry and dest to the default 
-                # registry.
-                registry = get_from_config('default_remote_registry')
-                if not registry:
-                    raise QuiltException("No registry specified and no default remote "
-                                        "registry configured. Please specify a "
-                                        "registry or configure a default remote "
-                                        "registry with quilt.config")
-
-                dest = registry
+            registry = get_from_config('default_remote_registry')
+            if not registry:
+                raise QuiltException("No registry specified and no default remote "
+                                     "registry configured. Please specify a "
+                                     "registry or configure a default remote "
+                                     "registry with quilt3.config")
+            registry_parsed = urlparse(fix_url(registry))
```

Comment:
Theoretically, the `default_local_registry` could be set to a `file://` URL, an S3 URL with a path, etc.

I'd move the checks below out of the `else` clause, and run them unconditionally.

Reply:
Good eye. But, in my opinion instead of doing this we should disallow setting `default_local_registry` to a non-conformant remote URL in the `quilt3.config` code, and assume that it is valid here. E.g. we should restrict `default_local_registry` to a local folder at set time.

Reply:
Ok, that's fine with me